pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -397,6 +404,19 @@ func (oe *operationExecutor) DetachVolume(
 		volumeToDetach.VolumeName, "" /* podName */, detachFunc)
 }
 
+func (oe *operationExecutor) CheckVolumesAreAttached(
+	attachedVolumes []AttachedVolume,
+	nodeName types.NodeName,
+	actualStateOfWorld ActualStateOfWorldAttacherUpdater) error {
+	volumesAreAttachedFunc, err :=
+		oe.generateVolumesAreAttachedFunc(attachedVolumes, nodeName, actualStateOfWorld)
+	if err != nil {
+		return err
+	}
+	// Give an empty UniqueVolumeName so that this operation could be executed concurrently.
+	return oe.pendingOperations.Run("", "", volumesAreAttachedFunc)
```

Comment:
What I'm trying to say is that since we are not trying to coordinate with other operations. We don't need to use either `nestedpendingoperations` `gorutineMap` or `operation_executor`. We can do all the things that the method returned by `generateVolumesAreAttachedFunc` does in thes same file as `syncstates` serially.